### ISSUE TITLE
Remove the `fully rebuild index` option from confirmation dialog of clear cache

### ISSUE THREAD

**Issue by Keith-CY** [Magickbase/neuron-public-issues#330]
The `fully rebuild index` option in the confirmation dialog of `clear cache` was used to remove data of the independent ckb-indexer before, by checking this, the ckb-indexer will re-index data to fix transaction history.

After ckb-indexer was merged into the ckb full node, `fully rebuild index` is intended to remove the `indexer` directory of the full node. This directory is a part of full node, and can be treated as consistent with full node data. If the `indexer` data is broken, the entire data should be flushed and can be done by **Menu => Tools => Clear all synchronized data**.

In conclusion, the `fully rebuild index` became meaningless when the independent ckb-indexer was removed, so the option can be removed now.

**Comment by Danie0918** [comment]
Consistent with the light client.

![image](https://github.com/Magickbase/neuron-public-issues/assets/122437870/ce490db3-bb11-4b7e-ace0-fb81ca17bfc3)

**Comment by Keith-CY** [comment]
> Consistent with the light client.
> 
> ![image](https://private-user-images.githubusercontent.com/122437870/288598537-ce490db3-bb11-4b7e-ace0-fb81ca17bfc3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIyNTM4NzgsIm5iZiI6MTcwMjI1MzU3OCwicGF0aCI6Ii8xMjI0Mzc4NzAvMjg4NTk4NTM3LWNlNDkwZGIzLWJiMTEtNGI3ZS1hY2UwLWZiODFjYTE3YmZjMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxMVQwMDEyNThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMmZiNGYzZDhiMmU3ZjVjMjE2ZjIzYWM1NzUxNjEyMGJiODhkMmQ1OTQ1YzI2ZDBiZmIwODI1MjQ3MzFhMzQ2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ILmXKtEnbAqdClkdyJZLu3oxnqzpJrlXrqdfBtCjyZw)

If there won't be more options or operations on `clear cache`, can we simplify the action by removing the dialog? Because this action takes less than 3 secs and doesn't have destructive results.

**Comment by Danie0918** [comment]
Checked out some of the software and for clearing the cache, most of them prompt a pop-up window. We can keep the pop-ups, but not use warning language to avoid the psychological burden on the user.

Text:It may take some time to clear the cache and rebuild the data, so please be patient.

![image](https://github.com/Magickbase/neuron-public-issues/assets/122437870/f9437e14-0880-4974-b0f8-bf03bab0f121)

@Keith-CY @yanguoyu

**Comment by Keith-CY** [comment]
> Checked out some of the software and for clearing the cache, most of them prompt a pop-up window. We can keep the pop-ups, but not use warning language to avoid the psychological burden on the user.
> 
> Text:It may take some time to clear the cache and rebuild the data, so please be patient.
> 
> ![image](https://private-user-images.githubusercontent.com/122437870/289452204-f9437e14-0880-4974-b0f8-bf03bab0f121.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIyODM3MTEsIm5iZiI6MTcwMjI4MzQxMSwicGF0aCI6Ii8xMjI0Mzc4NzAvMjg5NDUyMjA0LWY5NDM3ZTE0LTA4ODAtNDk3NC1iMGY4LWJmMDNiYWIwZjEyMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxMVQwODMwMTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZGJiNzQwMzRlZjNkZDg0ZGM2ZTNmZWY0MTllNjg1NDVkMjdmODBmOTQxMTcwNGY0ZTQwNjAyNWExMzU3YWUwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.yEWca8t9ypdGym6Otv01JEJ1Nx3qGL6eLuqSbIFlu58)
> 
> @Keith-CY @yanguoyu

I would like to know which applications show the alert/dialog on clearing cache. I ask this because I actually haven't seen an application(my daily used ones) does this.

I've tried
- Telegram
- Chrome
- Safari(ask for the duration)
- NetEaseMusic
- Apple Music

Didn't try WeChat because its cache strategy is weird, cannot be recovered after clearing cache

**Comment by yanguoyu** [comment]
Chrome will open a dialog to show which data will be removed and what effect it has. But it needs dialog because it has many options. Maybe we can add a tooltip for the `clear cache` button to show what will happen.
<img width="511" alt="image" src="https://github.com/Magickbase/neuron-public-issues/assets/12881040/69e0001c-00e2-499c-9383-f035816e2850">

**Comment by Danie0918** [comment]
> > Checked out some of the software and for clearing the cache, most of them prompt a pop-up window. We can keep the pop-ups, but not use warning language to avoid the psychological burden on the user.
> > Text:It may take some time to clear the cache and rebuild the data, so please be patient.
> > ![image](https://private-user-images.githubusercontent.com/122437870/289452204-f9437e14-0880-4974-b0f8-bf03bab0f121.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIyODM3MTEsIm5iZiI6MTcwMjI4MzQxMSwicGF0aCI6Ii8xMjI0Mzc4NzAvMjg5NDUyMjA0LWY5NDM3ZTE0LTA4ODAtNDk3NC1iMGY4LWJmMDNiYWIwZjEyMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxMVQwODMwMTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZGJiNzQwMzRlZjNkZDg0ZGM2ZTNmZWY0MTllNjg1NDVkMjdmODBmOTQxMTcwNGY0ZTQwNjAyNWExMzU3YWUwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.yEWca8t9ypdGym6Otv01JEJ1Nx3qGL6eLuqSbIFlu58)
> > @Keith-CY @yanguoyu
> 
> I would like to know which applications show the alert/dialog on clearing cache. I ask this because I actually haven't seen an application(my daily used ones) does this.
> 
> I've tried
> 
> * Telegram
> * Chrome
> * Safari(ask for the duration)
> * NetEaseMusic
> * Apple Music
> 
> Didn't try WeChat because its cache strategy is weird, cannot be recovered after clearing cache

Have tried X, Youtube, Instagram, and other apps in the phone, but didn't find the ability to clear cache, so it didn't count.

Chrome has a clear option prompt for the user when selecting, so it makes sense that there is no confirmation box, this can't be used as a reference.

Apps like AE, China Merchants Bank, JD mall, Taobao, etc. will pop up a prompt.

Facebook does not require a prompt.

**Comment by Danie0918** [comment]
Of course the list here is not comprehensive enough to conclude 'most', and that needs to be fixed.

However, there should be a reminder to clear the cache, either in the form of additional text or a pop-up window. This is because the user doesn't know if this action has any effect or not.

**Comment by Keith-CY** [comment]
> Of course the list here is not comprehensive enough to conclude 'most', and that needs to be fixed.
> 
> However, I still think that there should be a reminder to clear the cache, either in the form of additional text or a pop-up window. This is because the user doesn't know if this action has any effect or not.

So the point would be if it is worth informing users about `what does clear cache do` with a dialog/double confirmation.

As mentioned in https://github.com/Magickbase/neuron-public-issues/issues/330#issuecomment-1849583125, telling what will be removed and what will they result in makes sense because data are categorized, what is more important is that these data are unrecoverable.

But in Neuron, what `clear cache` does is simply refresh data, it does not actually remove something. Speaking of the result, it may take seconds to minutes to refresh the data. IMO, the only valuable information is `how long` it takes to refresh the data, the time estimation could be appended in label

**Comment by Danie0918** [comment]
> > Of course the list here is not comprehensive enough to conclude 'most', and that needs to be fixed.
> > However, I still think that there should be a reminder to clear the cache, either in the form of additional text or a pop-up window. This is because the user doesn't know if this action has any effect or not.
> 
> So the point would be if it is worth informing users about `what does clear cache do` with a dialog/double confirmation.
> 
> As mentioned in [#330 (comment)](https://github.com/Magickbase/neuron-public-issues/issues/330#issuecomment-1849583125), telling what will be removed and what will they result in makes sense because data are categorized, what is more important is that these data are unrecoverable.
> 
> But in Neuron, what `clear cache` does is simply refresh data, it does not actually remove something. Speaking of the result, it may take seconds to minutes to refresh the data. IMO, the only valuable information is `how long` it takes to refresh the data, the time estimation could be appended in label.
> 
> Maybe the word `clear cache` is misleading, it's more like `refresh the app`, it doesn't actually take something valuable away.

So under what circumstances does it take a user a few minutes to rebuild their data? Does it affect the user in the meantime?
@yanguoyu

**Comment by yanguoyu** [comment]
> > > Of course the list here is not comprehensive enough to conclude 'most', and that needs to be fixed.
> > > However, I still think that there should be a reminder to clear the cache, either in the form of additional text or a pop-up window. This is because the user doesn't know if this action has any effect or not.
> > 
> > 
> > So the point would be if it is worth informing users about `what does clear cache do` with a dialog/double confirmation.
> > As mentioned in [#330 (comment)](https://github.com/Magickbase/neuron-public-issues/issues/330#issuecomment-1849583125), telling what will be removed and what will they result in makes sense because data are categorized, what is more important is that these data are unrecoverable.
> > But in Neuron, what `clear cache` does is simply refresh data, it does not actually remove something. Speaking of the result, it may take seconds to minutes to refresh the data. IMO, the only valuable information is `how long` it takes to refresh the data, the time estimation could be appended in label.
> > Maybe the word `clear cache` is misleading, it's more like `refresh the app`, it doesn't actually take something valuable away.
> 
> So under what circumstances does it take a user a few minutes to rebuild their data? Does it affect the user in the meantime? @yanguoyu

The wallet transaction counts decide on the rebuilding time. Before the local data rebuild finishes, the user can not send ckb. For example, the test wallet has about 900 transactions, which will take about 10 minutes.

**Comment by Danie0918** [comment]
> > > > Of course the list here is not comprehensive enough to conclude 'most', and that needs to be fixed.
> > > > However, I still think that there should be a reminder to clear the cache, either in the form of additional text or a pop-up window. This is because the user doesn't know if this action has any effect or not.
> > > 
> > > 
> > > So the point would be if it is worth informing users about `what does clear cache do` with a dialog/double confirmation.
> > > As mentioned in [#330 (comment)](https://github.com/Magickbase/neuron-public-issues/issues/330#issuecomment-1849583125), telling what will be removed and what will they result in makes sense because data are categorized, what is more important is that these data are unrecoverable.
> > > But in Neuron, what `clear cache` does is simply refresh data, it does not actually remove something. Speaking of the result, it may take seconds to minutes to refresh the data. IMO, the only valuable information is `how long` it takes to refresh the data, the time estimation could be appended in label.
> > > Maybe the word `clear cache` is misleading, it's more like `refresh the app`, it doesn't actually take something valuable away.
> > 
> > 
> > So under what circumstances does it take a user a few minutes to rebuild their data? Does it affect the user in the meantime? @yanguoyu
> 
> The wallet transaction counts decide on the rebuilding time. Before the local data rebuild finishes, the user can not send ckb. For example, the test wallet has about 900 transactions, which will take about 10 minutes.

What happens on the front end is that the synchronization progress is reset and the data is rebuilt to restore the previous progress?
![image](https://github.com/Magickbase/neuron-public-issues/assets/122437870/50edc96f-cebc-455a-ae9d-b75537a0e49f)

**Comment by yanguoyu** [comment]
> What happens on the front end is that the synchronization progress is reset and the data is rebuilt to restore the previous progress?

It will grow from 0 to the lastest block number.

**Comment by yanguoyu** [comment]
Any update about this issue? @Danie0918  Has the interactive been confirmed yet?

**Comment by Danie0918** [comment]
> Any update about this issue? @Danie0918 Has the interactive been confirmed yet?

It's fine to keep the popup or not, but to make the product cleaner we remove it. And change "Clear Cache" to "Refresh". How about that? @ @Keith-CY @yanguoyu

**Comment by Keith-CY** [comment]
> > Any update about this issue? @Danie0918 Has the interactive been confirmed yet?
> 
> It's fine to keep the popup or not, but to make the product cleaner we remove it. And change the button "Clear Cache" to "Refresh". How about that? @ @Keith-CY @yanguoyu

LGTM

**Comment by yanguoyu** [comment]
https://github.com/nervosnetwork/neuron/pull/2990
